ami_assets/usr/lib/systemd/hugetlb-reserve-pages.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/sh
+#
+# Reserve hugepages when NUMA-aware.
+# Always reserve pages only on the first NUMA node.
+#
+# N.B. This currently only supports one or two NUMA nodes as it does not
+# NUMA nodes aren't guaranteed to be numbered contiguously and fully parsing
+# /sys/devices/system/node/online is non-trivial.
+#
+# NUMPAGES should be replaced with the required number of pages before use.
+nodes_path=/sys/devices/system/node/
+if [ ! -d "$nodes_path" ]; then
+	echo "ERROR: $nodes_path does not exist"
+	exit 1
+fi
+
+reserve_pages()
+{
+	echo "$1" > "$nodes_path/node$2/hugepages/hugepages-1048576kB/nr_hugepages"
```

Comment:
It's probably worth formatting this for consistency with other scripts in ami_assets (unless this is taken directly from somewhere and there are merge considerations?).

* 4 space indentation
* Functions like:

```
reserve_pages() {
    XXX
}